tsfc/fem.py

```diff
@@ -319,9 +317,10 @@ def callback(key):
         table = ctx.tabulation_manager[key]
         if len(table.shape) == 1:
             # Cellwise constant
-            row = gem.Literal(table)
+            row = table
         else:
-            table = ctx.index_selector(lambda i: gem.Literal(table[i]), mt.restriction)
+            table = ctx.index_selector(lambda i: gem.partial_indexed(table, (i,)),
```

Comment:
I think an `as_gem` akin to `as_ufl` would be right here. `table.array[i]` is a numpy array is `table` was a `Literal`, but a GEM expression if `table` was a `ListTensor`. Let me push a change.